proxy/src/outbound.rs

```diff
@@ -87,12 +88,20 @@ where
     fn recognize(&self, req: &Self::Request) -> Option<Reuse<Self::Key>> {
         let proto = bind::Protocol::detect(req);
 
-        let local = req.uri().authority_part().map(|authority| {
-            FullyQualifiedAuthority::normalize(
-                authority,
-                &self.default_namespace)
-
-        });
+        // The request URI and Host: header have not yet been normalized
+        // by `NormalizeUri`, as we need to know whether the request will
+        // be routed by Host/authority or by SO_ORIGINAL_DST, in order to
+        // determine whether the service is reusable.
```

Comment:
I think you should truncate this comment at `NormalizeUri` and then move the comment "// If we can't fully qualify the authority as a local service,.." here.